lib/puppet/provider/package/pkg.rb

```diff
@@ -160,6 +160,13 @@ def insync?(is)
         warning(_("Implicit version %{should} has %{n} possible matches") % { should: should, n: n })
       end
       potential_matches.each { |p|
+
+        # If the version that we match is installed then we don't need
+        # to check if it's installed (or installable), just return true
+        if (is != 'absent') && (p[:status] == 'installed')
```

Comment:
If the package is absent, then I think the `is` property will be the symbol `:absent` (so you'll want to do `is != :absent`)

Reply:
It does work as it stands. I believe that either way it maps to the string "absent". I can make that change if you really want it.

Reply:
I fixed this in the commit for the other change (and forgot to add it to the comment).